lib/plugin/git/Git.js

```diff
@@ -89,7 +89,10 @@ class Git extends GitBase {
   }
 
   isWorkingDirClean() {
-    return this.exec('git diff-index --quiet HEAD --', { options }).then(
+    // Note: the update-index is required here for cases where files may be touched/recreated
+    // during a build/test hook. Git will mark those as potentially changed, but won't diff them
+    // until the index is updated. See also https://github.com/release-it/release-it/issues/687
```

Comment:
I guess the comment can be removed again. Thanks for all the work! Will test this asap.

Reply:
whoops! removed.